backend/typescript/services/implementations/campService.ts

```diff
@@ -68,6 +66,33 @@ class CampService implements ICampService {
         return [];
       }
 
+      if (year) {
+        // Filter out camps that have camp sessions in the desired year
+        /* eslint-disable no-param-reassign */
+        camps = camps.filter((camp) => {
+          // Go through every camp session in a camp. If it has the desired year, include it in the results.
+          /* eslint-disable no-param-reassign */
+          camp.campSessions = (camp.campSessions as CampSession[]).filter(
+            (campSession) => {
+              return campSession.dates.every((campDate) => {
+                const startCampYearTime = new Date(year, 0, 1).getTime();
+                const endCampYearTime = new Date(year, 11, 31).getTime();
+                return (
+                  campDate.getTime() >= startCampYearTime &&
+                  campDate.getTime() <= endCampYearTime
+                );
+              });
+            },
+          );
+
+          // After filtering all the campSessions that have the wanted year, return all camps that have campSessions in the year or were created in the year
+          return (
+            camp.campSessions.length > 0 ||
+            new Date(camp.createdAt).getFullYear() === year
```

Comment:
ahh I see you used the function here. was there a reason you didn't use it for checking sessions?

Reply:
The code for checking sessions was taken from the original PR that was open for this lol I can update it

Reply:
yeah if it's the same thing, can you change it? I think this is cleaner. just verify that it works the same though